### PR TITLE
Use `lexicalContext` to improve diagnostics for tag declarations.

### PR DESCRIPTION
**Pull request by grynspan**
This PR uses the swift-syntax-600 `lexicalContext` property to enforce the compile-time requirement that tags declared with `@Tag` must be declared in an extension to `Tag` or in a type declared inside `Tag`. Fix-its are provided for a few scenarios. The new diagnostics are:

```swift
@Tag var x: Tag // 🛑 Attribute 'Tag' cannot be applied to a global variable
                // Declare in an extension to 'Tag' [Fix]
                // Remove attribute 'Tag' [Fix]

extension String {
  @Tag var x: Tag // 🛑 Attribute 'Tag' cannot be applied to a property except in an extension to 'Tag'
}

extension Tag {
  @Tag var x: Tag // 🛑 Attribute 'Tag' cannot be applied to an instance property
                  // Add 'static' [Fix]
}

extension Tag {
  @Tag var x: String // 🛑 Attribute 'Tag' cannot be applied to a property of type 'String'
                     // Change type to 'Tag' [Fix]
                     // Remove attribute 'Tag' [Fix]
}
```

As with #332 and #333, these problems are only diagnosed when using swift-syntax-600.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

File: Package.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ let package = Package(
   ],
 
   dependencies: [
-    .package(url: "https://github.com/apple/swift-syntax.git", from: "510.0.1"),
+    .package(url: "https://github.com/apple/swift-syntax.git", branch: "main"),
```
Comment (briancroom) [inline Package.swift:36]: This change would need to be reverted for a commit when tagging a release, right?

**Review by briancroom — COMMENTED** [review]
Neat!

File: Package.swift (hunk 1 above)
Comment (grynspan) [reply Package.swift:36]: I did not intend to include that in the PR. Will remove.

Comment (grynspan) [reply Package.swift]: Removed.

**Comment by grynspan** [comment]
@swift-ci please test

**Review by dennisweissmann — APPROVED** [review]

**Pull request merged**